evm/its.js

```diff
@@ -437,6 +463,36 @@ async function processCommand(_, chain, options) {
             break;
         }
 
+        case 'postDeploy': {
+            const interchainTokenService = new Contract(interchainTokenServiceAddress, InterchainTokenService.abi, wallet);
+
+            const contractConfig = chain.contracts[contractName];
+
+            const interchainTokenDeployer = await interchainTokenService.interchainTokenDeployer();
+            const interchainTokenFactory = await interchainTokenService.interchainTokenFactory();
+
+            const interchainTokenFactoryContract = new Contract(interchainTokenFactory, InterchainTokenFactory.abi, wallet);
+            const interchainTokenFactoryImplementation = await interchainTokenFactoryContract.implementation();
+
+            const interchainTokenDeployerContract = new Contract(interchainTokenDeployer, IInterchainTokenDeployer.abi, wallet);
+            const interchainToken = await interchainTokenDeployerContract.implementationAddress();
+
+            const toCheck = {
+                tokenManagerDeployer: await interchainTokenService.tokenManagerDeployer(),
+                interchainTokenDeployer,
+                interchainToken,
+                tokenManager: await interchainTokenService.tokenManager(),
+                tokenHandler: await interchainTokenService.tokenHandler(),
+                implementation: await interchainTokenService.implementation(),
+                interchainTokenFactory,
+                interchainTokenFactoryImplementation,
```

Comment:
should also check all constructor args used for deployment, and at least print those for manual verification. We get the constructor args in verify-contract script so you can create a helper maybe.

Reply:
added some additional checks